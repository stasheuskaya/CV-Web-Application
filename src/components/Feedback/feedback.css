.feedbackInfo {
  display: flex;
  align-items: center;
  padding-top: 18px;
  padding-bottom: 50px;
}

.feedbackName > a {
  color: var(--green-color-highlight-shamrock);
}

.feedback {
font-style: italic;
}

/* .feedbackAvatar {
  width: 50px;
  height: 50px;
  overflow: hidden;
  border-radius: 50%;
  margin-right: 20px;
} */

.loadingIndicator {
  color: var(--green-color-highlight-shamrock);
  width: 20px;
  height: 20px;
  animation:spin 2s linear infinite;
}

@keyframes spin { 
  100% { 
      -webkit-transform: rotate(360deg); 
      transform:rotate(360deg); 
  } 
}

.errorMessage {
  text-align: center;
  color: var(--red-error-color);
  font-weight: 400;
}