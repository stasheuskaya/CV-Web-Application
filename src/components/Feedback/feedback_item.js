import Info from "../Info/info";

export default function FeedbackItem (props) {
    return (
        <div className="feedbackContainer">
            <blockquote className="feedback">
                <Info text={props.feedback}/>
            </blockquote>
            <div className="feedbackInfo">
                {/* <img className="feedbackAvatar" src={props.photoUrl} alt="Reporter"/> */}
                <div className="feedbackName">
                    {props.name}, <a href={props.citeUrl}>{props.citeUrl}</a>
                </div>
            </div>
        </div>
    )
}
