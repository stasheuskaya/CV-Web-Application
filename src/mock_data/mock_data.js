export const expertisesData = [
  {
      date: '2021-2022', 
      info: {
        company: 'EPAM Systems inc',
        job: 'FrontEnd Developer/Engineer',
        description: 'Develop various web application, including the a Capstone project "CV Web application".  CV web application is a personal, professional CV web application that allows users to have a working prototype to share with audiences to introduce themselves.'
      }
  },
  {
      date: '2019-2021', 
      info: {
        company: 'Selfemployee',
        job: 'Mobile Software Testing Engineer  / QA',
        description: 'MyBudget mobile application (iOS). Set a budget and keep it with real time budget stats.'
      }
  },
  {
    date: '2017-2018', 
    info: {
      company: 'EPAM Systems inc',
      job: 'Mobile Software Testing Engineer  / QA',
      description: 'Ecommerce mobile application (iOS, Android). Global specialist in energy management and automation, with global presence in over 100 countries. It provides integrated efficiency solutions, combining energy, automation and software.'
    }
  },
  {
    date: '2015-2016', 
    info: {
      company: 'EPAM Systems inc',
      job: 'Software Testing Engineer',
      description: 'Ecommerce mobile application (iOS, Android) online-fashion outlet, with operations across Spain, Italy, Brazil and Mexico.'
    }
  },
  {
      date: '2014-2015', 
      info: {
        company: 'EPAM Systems inc',
        job: 'Software Testing Engineer',
        description: 'SaaS based Supply Chain and Industry platform, provides a set of Internet services for the automotive and manufacturing industry, with global presents in over 70 countries.'
      }
    }
]

export const feedbackData = [
  {feedback: '"Yulia is an incredibly talented, hard-working individual. I would recommend her to work in any area she pursues. She volunteered under me and also worked for a school I was on the board for. In both circumstances, paid and unpaid, she was dedicated, faithful, consistent, honest, kind, loving, full of good ideas, took initiative, and easy to work with."', 
  reporter: { photoUrl: 'https://www.atlassian.com/dam/jcr:ba03a215-2f45-40f5-8540-b2015223c918/Max-R_Headshot%20(1).jpg', 
  name: 'Allyse Tanovitz, YVCS School board member', 
  citeUrl: 'https://yvcschool.org' } 
  }, 

  {feedback: '"Yuliya was a valuable member of our team, and her job and attitude helped us to finish project successfully. Yuliya proved herself not only as a good QA engineer, but also as a consistent and reliable person, it was easy and comfortable to work with Yuliya."'
  , 
  reporter: 
  { photoUrl: 'https://www.atlassian.com/dam/jcr:ba03a215-2f45-40f5-8540-b2015223c918/Max-R_Headshot%20(1).jpg', 
  name: 'Marina Tretjakova, QA Test Lead, EPAM Systems', 
  citeUrl: 'www.epam.com' } } 
]

export const timeLineData = [
{
  "date": 2022,
  "title": "‘Front-end Development‘ technology bootcamp",
  "text": "‘Front-end Development‘ EPAM UpSkill technology bootcamp, USA: IT Fundamentals; HTML and CSS; JavaScript Basics, Advanced, TypeScript, React, "
},
{
  "date": 2015,
  "title": "‘Mobile testing’ training",
  "text": "‘Mobile testing’ training, EPAM Systems inc, Belarus: iOS, Android, Windows mobile testing; Mobile specific testing; API testing (Charles); Crash Logs tracking (Xcode, Android SDK)."
},
{
  "date": 2014,
  "title": "‘Software Testing introduction‘ training",
  "text": "‘Software Testing introduction‘ training, EPAM Systems inc, Belarus. Manual web application testing; Automation testing (Selenium), Performance testing (Tomcat)."
},
{
  "date": 2008,
  "title": "Bachelor in Business Activities",
  "text": " ’Institute of Entrepreneurial Activity’, Belarus."
}
]

export const portfolioData = [
{
  "type": "ui",
  "title": "‘Front-end Development‘ technology bootcamp",
  "text": "‘Front-end Development‘ EPAM UpSkill technology bootcamp, USA: IT Fundamentals; HTML and CSS; JavaScript Basics, Advanced, TypeScript, React, "
},
{
  "type": "code",
  "title": "‘Mobile testing’ training",
  "text": "‘Mobile testing’ training, EPAM Systems inc, Belarus: iOS, Android, Windows mobile testing; Mobile specific testing; API testing (Charles); Crash Logs tracking (Xcode, Android SDK)."
},
{
  "type": "ui",
  "title": "‘Software Testing introduction‘ training",
  "text": "‘Software Testing introduction‘ training, EPAM Systems inc, Belarus. Manual web application testing; Automation testing (Selenium), Performance testing (Tomcat)."
},
{
  "type": "code",
  "title": "Bachelor in Business Activities",
  "text": " ’Institute of Entrepreneurial Activity’, Belarus."
}
]

export const skillsData = [
{
  "name": "React",
  "range": 10
},
{
  "name": "JS",
  "range": 100
}
]