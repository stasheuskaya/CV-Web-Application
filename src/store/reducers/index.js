import { combineReducers } from "redux";
import { skillsReducer } from "./skills_reducer";
import { educationsReducer } from "./educations_reducer";

const allReducers = combineReducers({
    skills: skillsReducer,
    educations: educationsReducer
})

export default allReducers;