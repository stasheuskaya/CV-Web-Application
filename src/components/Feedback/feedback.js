import './feedback.css';
import FeedbackItem from "./feedback_item";

export default function Feedback(props) {
    return (
      <div>
        {props.data.map(item => 
          <FeedbackItem
            feedback={item.feedback}
            key={item.feedback}
            // photoUrl={item.reporter.photoUrl} 
            name={item.reporter.name} 
            citeUrl={item.reporter.citeUrl}
          />)}
      </div>
    )
}
