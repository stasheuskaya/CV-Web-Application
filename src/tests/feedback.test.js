import renderer from "react-test-renderer";
import Feedback from "../components/Feedback/feedback";
import { Provider } from 'react-redux';
import { store } from "../store/store"
import { feedbackData } from "../mock_data/mock_data"

describe('Feedback Page', () => {
    it('renders content', () => {
		expect(renderer.create(
			<Provider store={store}>
                <Feedback data={feedbackData}/>
			</Provider>)).toMatchSnapshot();
	});
});