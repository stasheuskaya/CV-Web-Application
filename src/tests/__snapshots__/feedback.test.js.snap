// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback Page renders content 1`] = `
<div>
  <div
    className="feedbackContainer"
  >
    <blockquote
      className="feedback"
    >
      <div
        className="infoBox"
      >
        "Yulia is an incredibly talented, hard-working individual. I would recommend her to work in any area she pursues. She volunteered under me and also worked for a school I was on the board for. In both circumstances, paid and unpaid, she was dedicated, faithful, consistent, honest, kind, loving, full of good ideas, took initiative, and easy to work with."
      </div>
    </blockquote>
    <div
      className="feedbackInfo"
    >
      <div
        className="feedbackName"
      >
        Allyse Tanovitz, YVCS School board member
        , 
        <a
          href="https://yvcschool.org"
        >
          https://yvcschool.org
        </a>
      </div>
    </div>
  </div>
  <div
    className="feedbackContainer"
  >
    <blockquote
      className="feedback"
    >
      <div
        className="infoBox"
      >
        "Yuliya was a valuable member of our team, and her job and attitude helped us to finish project successfully. Yuliya proved herself not only as a good QA engineer, but also as a consistent and reliable person, it was easy and comfortable to work with Yuliya."
      </div>
    </blockquote>
    <div
      className="feedbackInfo"
    >
      <div
        className="feedbackName"
      >
        Marina Tretjakova, QA Test Lead, EPAM Systems
        , 
        <a
          href="www.epam.com"
        >
          www.epam.com
        </a>
      </div>
    </div>
  </div>
</div>
`;
